libsolidity/CMakeLists.txt

```diff
@@ -27,7 +27,8 @@ else()
 endif()
 
 if (NOT (${Z3_FOUND} OR ${CVC4_FOUND}))
-  message("No SMT solver found. Optional SMT checking will not be available. Please install Z3 or CVC4 if it is desired.")
+  message("No SMT solver found or USE_ flags have been set. Optional SMT checking will not be available.\
+  \nPlease install Z3 or CVC4 if it is desired or remove USE_ flags and re-run cmake without them.")
```

Comment:
I don't think the "remove USE_ flags" part is explaining it well. Perhaps: "No SMT solver found (or it has been forcefully disabled). Optional SMT checking will not be available. Please install Z3 or CVC4 or remove the option disabling them (USE_Z3, USE_CVC4)."